.github/workflows/integration.yml

```diff
@@ -44,7 +44,7 @@ jobs:
         sudo python -m pip install pipenv
         # pipenv is not working reliably when combined with sudo and a python_version requirement
         sudo pipenv requirements > requirements.txt
-        sudo python -m pip install -r requirements.txt
+        sudo python -m pip install -r requirements.txt --use-pep517
```

Comment:
Can you give a tl;dr version of what ``--use-pep517` does and why we use it for a non-pythonista?

Reply:
It seems like some of the salt dependencies (`pyzmq`) have some issues and this is a workaround and uses a more modern approach to build tools 

```
      /usr/local/lib/python3.10/dist-packages/setuptools/__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
        dist.fetch_build_eggs(dist.setup_requires)
      
      
              It is possible a package already installed in your system
              contains an version that is invalid according to PEP 440.
              You can try `pip install --use-pep517` as a workaround for this problem,
              or rely on a new virtual environment.
      
              If the problem refers to a package that is not installed yet,
              please contact that package's maintainers or distributors.
```